### ISSUE TITLE
Rule proposal: `prefer-logical-assignment-operators`

### ISSUE THREAD

**Issue by dimaMachina** [sindresorhus/eslint-plugin-unicorn#1503]
Less code written, more readable
## Fail

```js
obj[foo] = obj[foo] || 'bad'

obj[foo] = obj[foo] && 'bad'

obj[foo] = obj[foo] ?? 'bad'
```

## Pass

```js
obj[foo] ||= 'good'

obj[foo] &&= 'good'

obj[foo] ??= 'good'
```

**Comment by dasa** [comment]
Rule has been added at: https://eslint.org/docs/latest/rules/logical-assignment-operators
